src/Tizen.Applications.ComponentBased/Tizen.Applications.ComponentBased.Common/ComponentFactoryBase.cs

```diff
@@ -0,0 +1,131 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Tizen.Applications.CoreBackend;
+using static Interop.CBApplication;
+using static Tizen.Applications.CoreBackend.DefaultCoreBackend;
+
+namespace Tizen.Applications.ComponentBased.Common
+{
+    internal abstract class ComponentFactoryBase
```

Comment:
I think `BaseComponentFactory` is better

BaseComponentFactory
FrameComponentFactory : BaseComponentFactory
ServiceComponentFactory : BaseComponentFactory

Because, class hierarchy should be matched
BaseComponent
FrameComponent : BaseComponent
ServiceComponent : BaseComponent

Reply:
And this class is not factory
It violation of single responsibility principle

Reply:
Why do you think it is not an Abstract Factory pattern?
In my opinion, it can be an Abstract Factory pattern's factory.
It provides Bind function so that the ComponentBasedApplication can create instances of Components.
This class allows ComponentBasedApplication to create concreate FrameComponent or ServiceComponent without dependency.

> 
https://en.wikipedia.org/wiki/Abstract_factory_pattern

The client doesn't know (or care) which concrete objects it gets from each of these internal factories, since it uses only the generic interfaces of their products

> 

And about single responsibility, if you agree that it can be an Abstract Factory pattern,
this class does not violate the single responsibility since it providing a contract for the creation of polymorphic objects without binding the interface to a specific implementation.

https://stackoverflow.com/questions/46790893/abstract-factory-violate-single-responsibility-principle

Reply:
It violation of single responsibility principle
Absract Factory class only create a some object, but your code dispatch event

Reply:
I never seen Factory pattern without create method.

Reply:
It look like state pattern.  However, I don't know why it violate SRP. BaseComponentFactory just handles events. Because it is internal classes, it is not relate to ACR. why don't you discuss it later?

Reply:
Anyway, it is not factory, please rename

Reply:
Well, I thought Bind function create objects indirectly, and thus it could be an abstract factory pattern. Also, native-c implementation using a standard factory pattern for this feature, so it made me confusing I guess.

Anyway, I will fix it. How about BaseComponentStateManger than?

Reply:
Yes, `BaseComponentStateManager` is goog to understand. 
FrameComponentStateManager : ComponentStateManger
ServiceComponentStateManager : ComponentStateManger

And I want to match a hierarchy between Component and this

Reply:
What do you mean to match a hierarchy between Component and this?
Isn't it already have the same hierarchy? Do I have to change something?